### PR TITLE
138-ADMIN-PROMOTION-Endpoint: Superadmin only

### PR DESCRIPTION
**Pull request by JAcciarri**

### CODE REVIEW

File: create-test-env-file.sh [hunk 1]
```diff
@@ -29,4 +29,6 @@ echo EMAIL_CONFIRMATION_URL="$FRONTEND_HOST/account/verifyEmail" >> $ENV_FILE
 echo SENDGRID_API_KEY="SG.example" >> $ENV_FILE
 echo SENDINBLUE_API_KEY="example" >> $ENV_FILE
 echo SAME_SITE_POLICY="none" >> $ENV_FILE
-echo SECURE_COOKIE="true" >> $ENV_FILE
\ No newline at end of file
+echo SECURE_COOKIE="true" >> $ENV_FILE
+echo SUPER_ADMIN_EMAIL="$SUPER_ADMIN_EMAIL" >> $ENV_FILE
+echo SUPER_ADMIN_PASSWORD="$SUPER_ADMIN_PASSWORD" >> $ENV_FILE
```
Comment (alereca) [inline create-test-env-file.sh:34]: I think you should use example values here as this is only required for tests

File: src/user/user.controller.ts [hunk 2]
```diff
@@ -1,9 +1,18 @@
...
+  ParseIntPipe,
+  Post,
+  UseGuards,
+} from '@nestjs/common';
+import { ApiCookieAuth, ApiTags } from '@nestjs/swagger';
 import { UserService } from './user.service';
 import { AppValidationPipe } from '../utils/validation.pipe';
 import { UserFindDto } from './dtos/user.find.dto';
 import { UsersResult } from './dtos/user.show.dto';
+import { IsSuperAdminGuard } from '../auth/is.super.admin.guard';
```
Comment (alereca) [inline src/user/user.controller.ts:15]: Should we skip this superadmin user in UserController,get method?

File: test/user/user.e2e-spec.ts [hunk 3]
```diff
@@ -48,4 +48,19 @@ describe('User Actions (e2e)', () => {
...
+      const accessTokenCookie = res.header['set-cookie'][0];
+      await request(app.getHttpServer())
+        .post('/users/1/promote')
+        .set('Cookie', accessTokenCookie)
+        .then((res) => {
+          expect(res.status).toBe(401);
+          expect(res.body.message).toEqual('Unauthorized');
+        });
+    });
+  });
```
Comment (alereca) [inline test/user/user.e2e-spec.ts:65]: Great!

File: src/user/user.controller.ts (hunk 2 above)
Comment (JAcciarri) [reply src/user/user.controller.ts:15]: done in new changes pushed

**Review by alereca — APPROVED** [review]

**Pull request merged**